### PR TITLE
AdSpirit Bid Adapter

### PR DESCRIPTION
**Pull request by SpreeGorilla**
## Type of change
- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other

## Description of change
adding new bid adapter for AdSpirit

- test parameters for validating bids
```
{
  bidder: 'adspirit',
  params: {
    placementId: XXX,
    host: URL    // advertiser specific adserver url    
  }
}
```

```
{
  bidder: 'connectad',
  params: {
    placementId: XXX,
  }
}
```

```
{
  bidder: 'xapadsmedia',
  params: {
    placementId: XXX,
  }
}
```

- contact email of the adapter’s maintainer: github@spreegorillas.de
- [x]  official adapter submission

## Other information
related docs on https://github.com/prebid/prebid.github.io/pull/421

I'm not sure why Github finds changes in modules/fidelityBidAdapter.js and modules/kummaBidAdapter.js, since there aren't any changes actually. These files are irrelevant for this PR anyway.

### CODE REVIEW

File: modules/adspiritBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      return [];
+    }
+    var cpm = serverResponse.cpm;
+    if (!cpm) return [];
+
+    var host = spec.getBidderHost(bidObj);
+    var adm = '<scr' + 'ipt>window.inDapIF=false</scr' + 'ipt><scr' + 'ipt src="//' + host + SCRIPT_URL + '"></scr' + 'ipt>' + '<ins id="' + bidObj.adspiritConId + '"></ins>' + serverResponse.adm;
+    const bidResponse = {
+      requestId: bidRequest.bidId,
+      bidderCode: spec.code,
```
Comment (matthewlane) [inline modules/adspiritBidAdapter.js]: `bidderCode` will be set automatically now, this line can be dropped

File: modules/adspiritBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import * as utils from 'src/utils';
+import {registerBidder} from 'src/adapters/bidderFactory';
+const BIDDER_CODE = 'adspirit';
+const RTB_URL = '/rtb/getbid.php?rtbprovider=prebid';
+const SCRIPT_URL = '/adasync.min.js';
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: ['xapadsmedia'],
+  isBidRequestValid: function(bid) {
+    var host = spec.getBidderHost(bid);
```
Comment (matthewlane) [inline modules/adspiritBidAdapter.js]: Prefer `const` or `let` over `var` here and throughout rest of PR

File: modules/adspiritBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      requestId: bidRequest.bidId,
+      bidderCode: spec.code,
+      cpm: cpm,
+      width: serverResponse.w,
+      height: serverResponse.h,
+      creativeId: serverResponse.placement_id,
+      currency: 'EUR',
+      netRevenue: true,
+      ad: adm
+    };
```
Comment (matthewlane) [inline modules/adspiritBidAdapter.js]: This object should also set a `ttl`, see table in http://prebid.org/dev-docs/bidder-adapter-1.html#interpreting-the-response for description of this field

File: test/spec/modules/adspiritBidAdapter_spec.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
+import {expect} from 'chai';
+import {spec} from 'modules/adspiritBidAdapter';
+import * as utils from 'src/utils';
```
Comment (matthewlane) [inline test/spec/modules/adspiritBidAdapter_spec.js]: Looks like `utils` isn't used in the test, line can be dropped

File: modules/adspiritBidAdapter.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      netRevenue: true,
+      ttl: 300,
+      ad: adm
+    };
+    bidResponses.push(bidResponse);
+    return bidResponses;
+  },
+  getBidderHost: function(bid) {
+    if (bid.bidder === 'adspirit') return utils.getBidIdParameter('host', bid.params);
+    if (bid.bidder === 'connectad') return 'connected-by.connectad.io';
```
Comment (matthewlane) [inline modules/adspiritBidAdapter.js]: if `connectad` is an alias for this bidder it should also be added to the `aliases` array on line 8

File: modules/adspiritBidAdapter.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    var adUnits = [
+           {
+               code: 'display-div',
+               sizes: [[300, 250]],  // a display size
+               bids: [
+                   {
+                       bidder: "adspirit",
+                       params: {
+                           placementID: '123',
+                           host: 'host'
```
Comment (matthewlane) [inline modules/adspiritBidAdapter.md]: Are these parameters valid for testing bid responses?

**Comment by mkendall07** [comment]
@SpreeGorilla this needs updates and conflicts resolved.

**Comment by SpreeGorilla** [comment]
I deleted everything but the relevant files but now the Travis CI build fails (of course). Did I misunderstand?
In the test parameters there was just a typo.
Thanks for your support and patience!

**Comment by matthewlane** [comment]
@SpreeGorilla There are still a lot of conflicts, you may want to open a new PR based off of latest master with just your adapter changes added in so that we can properly review

**Comment by SpreeGorilla** [comment]
Travis CI build failed because "Cannot start ChromeHeadless". I'd guess that's not my fault? Not sure what to do here.

**Comment by mkendall07** [comment]
yeah travis seems to be broken. We'll take a look at it

File: integrationExamples/gpt/pbjs_ucfunnel_gpt.html [hunk 7]
```diff
@@ -1,97 +1,97 @@
...
-<!-- /18087395/header-bid-tag-0 -->
-<div id='div-gpt-ad-1495530934718-0'>
-    <script>
-        googletag.cmd.push(function() { googletag.display('div-gpt-ad-1495530934718-0'); });
-    </script>
-</div>
-
-</body>
-</html>
+<!DOCTYPE html>
```
Comment (mkendall07) [inline integrationExamples/gpt/pbjs_ucfunnel_gpt.html]: please drop changes to files that are not needed in this pr

**Comment by mkendall07** [comment]
also if you want travis to pass you have to rebase off master

**Comment by SpreeGorilla** [comment]
No sweat, but if you need me to do anything else here please let me know.

**Comment by bretg** [comment]
@SpreeGorilla - please resolve code conflicts

**Comment by mkendall07** [comment]
This pull request **fixes 4 alerts** when merging acdfc4096771dfe2cf6fca7b7a20705f20f9ba52 into 95fe0a58dca508c6c60da03e55bf28c1d0d1780b - [view on lgtm.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-51f65455653965013a0aa0bbe18a0a0aff411e7e)

**fixed alerts:**

* 4 for Identical operands

---

*Comment posted by [lgtm.com](https://lgtm.com)*

**Comment by SpreeGorilla** [comment]
So does this mean we are making progress here? Anything I need to do at the moment?

**Comment by mkendall07** [comment]
@SpreeGorilla 
You haven't addressed all our review comments. You don't need to commit this file for example: 
integrationExamples/gpt/pbjs_ucfunnel_gpt.html

**Comment by mkendall07** [comment]
This pull request **fixes 6 alerts** when merging 5a0f455acacfb864dc3f6a84d7e21d4cb27163a8 into f7813d509a023ba0e67e70266f1c94b29f6dfd7b - [view on lgtm.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-1da911be8fd5b6f3bf6e10dd29c57be230c53148)

**fixed alerts:**

* 4 for Identical operands
* 1 for Variable not declared before use
* 1 for Syntax error

---

*Comment posted by [lgtm.com](https://lgtm.com)*

**Comment by mkendall07** [comment]
This pull request **fixes 4 alerts** when merging 0ea4986d142c6452abf77696f134622a73373121 into 01decf4dac6863989a0a900e964a5c5adad40704 - [view on lgtm.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-b0f6093a2a660fc93ee574137c663dc2556edaa8)

**fixed alerts:**

* 2 for Client side cross\-site scripting
* 1 for Variable not declared before use
* 1 for Syntax error

---

*Comment posted by [lgtm.com](https://lgtm.com)*

**Comment by SpreeGorilla** [comment]
I'm not sure why there are still 2 requested changes. Are they preventing making progress here?  I fixed that quite a while ago. Please tell me if I can do something to speed things up!

**Comment by mkendall07** [comment]
closing because you haven't made the requested changes and this PR completely out of date. Open a new one with a fresh cut of master if you would like

**Pull request closed without merging**